### PR TITLE
Feature: Add support for build profile

### LINKED ISSUE #550: Add support for Build Profile and Host Profile
### Description of Problem

Hi we are using conan for building applications on ARM devices. This requires sometimes to set a Host- and Build-Profile for the package build. It would be nice if this feature is supported by the conan-package-tools.

### Environment Details
* Conan Package Tools Version: **0.23.0**
* Operating System: **Linux**
* Operation System Version: **Ubuntu 20.04**
* Compiler+version: **arm-noneabi-gcc-10**
* Conan version: **conan 1.33.0**
* Python version: **python 3.7.0**

### PR DESCRIPTION
**Pull request by ghost**
Changelog: (Feature): Add support for build profile

The build profile is passed to cpt.packager.run() using the
base_profile_build_name parameter.

Issue #550 

When running in Docker the complete build profile is passed to the
container using an environment variable and will there be stored in a
temporary file.

Unfortunately the current implementation does not allow passing
additional options, settings or environment variables to the build
profile like it is done for 'normal' profiles. This feature is not yet
supported by cpt.packager.ConanMultipackager.add().
But maybe this is a start for a discussion on how to implement this feature.

Host profiles could be added in a similar way.

- [X] Refer to the issue that supports this Pull Request.
- [X] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [X] I've read the [Contributing guide](https://github.com/conan-io/conan-package-tools/blob/develop/.github/CONTRIBUTING.md).
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008) style guides for Python code.

### CODE REVIEW

**Comment by datalogics-kam** [comment]
This PR would greatly benefit us as we could use the build profile for building the standard matrix of our packages. Is there anything holding up it being merged?

**Comment by uilianries** [comment]
Hello! It's a great add indeed, but this PR is still missing functional test. I'll check later to add some tests.

**Comment by ghost** [comment]
Good to get some feedback. I'll try to add some unit tests in the next week. I would also like to rename some of the variables I added to keep them aligned with the existing ones.

File: cpt/runner.py [hunk 1]
```diff
@@ -125,14 +131,21 @@ def run(self):
                                                         not_export=self.skip_recipe_export,
                                                         update=self._update_dependencies)
                             else:
+                                if self._profile_build_abs_path is not None:
+                                    profile_build = ProfileData([self._profile_build_abs_path],
```
Comment (dvirtz) [inline cpt/runner.py]: `ProfileData` has 5 fields but only 4 are set
https://github.com/conan-io/conan/blob/addd8ab4726a557d165e17ec6bb202f0c1666f0c/conans/client/conan_api.py#L74

**Review by dvirtz — COMMENTED** [review]
I've started using this branch for cross building from Linux to Windows with mingw and it works great.

File: cpt/runner.py (hunk 1 above)
Comment (ghost) [reply cpt/runner.py]: This branch has been implemented using v1.35 of Conan
https://github.com/conan-io/conan/blob/release/1.35/conans/client/conan_api.py#L74

In that version only 4 values were required.
I'll look into that today.

**Comment by uilianries** [comment]
Yesterday I did an upgrade on develop branch, please, don't forget to rebase your changes.

File: cpt/runner.py [hunk 2]
```diff
@@ -125,14 +131,21 @@ def run(self):
...
+                                                                None, None, None)
```
Comment (uilianries) [inline cpt/runner.py]:
```suggestion
                                    profile_build = ProfileData(profiles=[self._profile_build_abs_path], settings=None, options=None, env=None, conf=None)
```

Use parameters when using Conan API, as there is no contract which ensure the method signature.

**Comment by uilianries** [comment]
@smartSBA I did some changes in your PR, including new tests, so we can merge it soon. I want to release a new version this week.

**Comment by uilianries** [comment]
I'll merge it tomorrow, in case you want to comment something before.

**Comment by ghost** [comment]
Thanks for cleaning up the code and adding those tests.
There will be no more changes from my side. Looking forward to the new version.

**Pull request merged**